### PR TITLE
(core): convert url builder service to TS

### PR DESCRIPTION
**Pull request by icfantv**

### CODE REVIEW

**Comment by icfantv** [comment]
@anotherchrisberry @zanthrash PTAL

File: app/scripts/modules/core/navigation/urlBuilder.service.ts [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+  name: string;
+  project: string;
+  provider: string;
+  region: string;
+  serverGroup: string;
+  stack: string;
+  taskId: string;
+  type: string;
+  vpcId: string;
+}
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/navigation/urlBuilder.service.ts]: Aside from `type` and `name` (and possibly including `type` and `name`), all of these fields are probably optional.

**Comment by anotherchrisberry** [comment]
Aside from the optional fields, LGTM. Feel free to merge and we can mark the optional fields optional as needed when we convert the components relying on the UrlBuilder to TS.

**Comment by icfantv** [comment]
sounds good to me.

**Pull request merged**